cache.js

```diff
@@ -1,77 +1,118 @@
 /**
- * Shared cache namespace.
- * This namespace provides a default cache that can be used to set and retrieve objects that will be persisted between reloads of scripts.
- *
+ * Cache namespace.
+ * This namespace provides caches that can be used to set and retrieve objects that will be persisted between reloads of scripts.
  * @namespace cache
  */
 
-const cache = require('@runtime').sharedcache;
+const log = require('./log')('cache');
+const { privateCache, sharedCache } = require('@runtime/cache'); // The new cache from core
+const addonSharedCache = require('@runtime').sharedcache; // The old cache from the adddon
 
-/**
- * Returns the value to which the specified key is mapped
- *
- * @example <caption>Get a previously set value with a default value (times = 0)</caption>
- * let counter = cache.get("counter", () => ({ "times": 0 }));
- * console.log("Count",counter.times++);
- *
- * @example <caption>Get a previously set object</caption>
- * let counter = cache.get("counter");
- * if(counter == null){
- *      counter = {times: 0};
- *      cache.put("counter", counter);
- * }
- * console.log("Count",counter.times++);
- *
- * @memberof cache
- * @param {string} key the key whose associated value is to be returned
- * @param {function} [defaultSupplier] if the specified key is not already associated with a value, this function will return a default value
- * @returns {(*|null)} the current object for the supplied key, a default value if defaultSupplier is provided, or null
- */
-const get = function (key, defaultSupplier) {
-  if (typeof defaultSupplier === 'function') {
-    return cache.get(key, defaultSupplier);
-  } else {
-    return cache.get(key);
-  }
-};
+const coreCacheAvail = Java.isJavaObject(privateCache) && Java.isJavaObject(sharedCache);
 
-/**
- * Associates the specified value with the specified key
- *
- * @memberof cache
- * @param {string} key key with which the specified value is to be associated
- * @param {*} value value to be associated with the specified key
- * @returns {(*|null)} the previous value associated with null, or null if there was no mapping for key
- */
-const put = function (key, value) {
-  return cache.put(key, value);
+const logDeprecationWarning = (funcName) => {
```

Comment:
Is there any way we can get the ruleUID or filename to log out as part of this warning? At first some users may get hundreds or these warnings and knowing which rules they are coming from can help in prioritization in dealing with them.

Reply:
By using `console.warn` instead of `log.warn` we are logging to the logger name of the script/rule, so for file-based script to `org.openhab.automation.script.file.filename.js` and for UI to `org.openhab.automation.script.ui.ruleUid`.